OpenRA.Mods.Common/Installer/SourceActions/DeleteSourceAction.cs

```diff
@@ -0,0 +1,29 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace OpenRA.Mods.Common.Installer
+{
+	public class DeleteSourceAction : ISourceAction
+	{
+		public void Run(MiniYaml actionYaml, string path, ModData modData, List<string> extracted, Action<string> updateMessage)
+		{
+			// Yaml path may be specified relative to a named directory (e.g. ^SupportDir) or the detected source path
+			var sourcePath = actionYaml.Value.StartsWith("^") ? Platform.ResolvePath(actionYaml.Value) : Path.Combine(path, actionYaml.Value);
```

Comment:
What is the usecase for deleting files in the source path? The source isn't an OpenRA-owned directory, so we can't assume we will have permissions to do this!

Reply:
Well... i didnt invent this :D It was already here. See: https://github.com/OpenRA/OpenRA/blob/bleed/mods/cnc/installer/firstdecade.yaml#L336
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/installer/firstdecade.yaml#L32
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/installer/firstdecade.yaml#L448
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ts/installer/firstdecade.yaml#L58
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ts/installer/firstdecade.yaml#L594

Reply:
I mean cases that don't start with the ^!

Reply:
Oh! Okay, ill add a change for this :)